docs/design/CONFIG_API.md

```diff
@@ -0,0 +1,231 @@
+	// +kubebuilder:validation:Enum=XdsServer;Provisioner
+	// +kubebuilder:default=XdsServer;Provisioner
+	EnabledServices []ServiceType `json:"enabledServices,omitempty"`
+
+	// XDSServer contains parameters of the xDS server. If unspecified,
+	// the xDS server will use default configuration settings.
+	//
+	// +optional
+	XDSServer *XDSServerConfig `json:"xdsServer,omitempty"`
+
+	// Provisioner contains parameters of the provisioner server. If
+	// unspecified, the provisioner server will use default configuration
+	// settings.
+	//
+	// +optional
+	Provisioner *ProvisionerConfig `json:"Provisioner,omitempty"`
+}
+
+// ServiceType defines the type of services supported by Envoy Gateway.
+//
+// +kubebuilder:validation:Enum=XdsServer;Provisioner
+type ServiceType string
+
+const (
+	// XdsServerServiceType is the name of the "XdsServer" service type.
+    XdsServerServiceType ServiceType = "XdsServer"
+
+	// ProvisionerServiceType is the name of the "Provisioner" service type.
+    ProvisionerServiceType ServiceType = "Provisioner"
+)
+
+// XDSServerConfig contains the configuration of the xDS server.
+type XDSServerConfig struct {
+	// Defines the IP address of the server to serve xDS over gRPC.
+	//
+	// If unspecified, defaults to "0.0.0.0", e.g. all IP addresses.
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:default=0.0.0.0
+	Address *string `json:"address,omitempty"`
+
+	// Defines the network port for the xDS server to serve xDS over gRPC.
+	//
+	// If unspecified, defaults is 18000.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:default=18000
+	Port *int32 `json:"port,omitempty"`
+}
+
+// ProvisionerConfig defines the provisioner server configuration.
+type ProvisionerConfig struct {
+	// ControlPlane specifies deployment configuration of the control plane,
+	// e.g. the xDS server.
+	//
+	// +optional
+	ControlPlane *ControlPlaneConfig `json:"controlPlane,omitempty"`
+
+	// Envoy specifies deployment configuration of the managed Envoy data
+	// plane.
+	//
+	// +optional
+	Envoy *EnvoyConfig `json:"envoy,omitempty"`
+}
+
+// ControlPlaneConfig defines the configuration of the control plane deployment.
+type ControlPlaneConfig struct {
+	// Replicas is the desired number of Contour replicas. If unspecified,
+	// defaults to 2.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=2
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// NetworkPublishing defines how to publish network endpoint(s) to a network.
+	// If unspecified, defaults to "KubeLoadBalancerService" which defines a
+	// Kubernetes service of type LoadBalancer.
+	//
+	// +optional.
+	NetworkPublishing *NetworkPublishing `json:"networkPublishing,omitempty"`
+}
+
+// EnvoyConfig defines the configuration of the Envoy data plane deployment.
+type EnvoyConfig struct {
+	// Replicas is the desired number of Envoy replicas. If unspecified,
```

Comment:
Let's be more specific and use `Envoy Proxy` in place of `Envoy`.

Reply:
@alexgervais please see https://github.com/envoyproxy/gateway/pull/95/files#r891594239.